### DISCUSSION TITLE
0x0B error code

### DISCUSSION THREAD

**Discussion by DIEGGOMEZ** [Infineon/optiga-trust-m#126] [Q&A]
Hi,

I am trying to decrypt a message, and the returned code is 0x0B 'Out of sequence', can you give me more information about this error?

Thank you.

**Comment by sss551** [comment]
Hi @DIEGGOMEZ ,

Can you please provide which API resulted in this error ?

Best regards,
Shiva Sai

**Reply by DIEGGOMEZ** [reply]
Of course, i am developing in Keil, debugging an STM32, trying to decrypt:

		                optiga_status = OPTIGA_LIB_BUSY;
		                optiga_status = optiga_util_open_application(util,0);
		                util = optiga_util_create(OPTIGA_INSTANCE_ID_0,optiga_util_callback,NULL); //OPTIGA_INSTANCE_ID_0
	//Decrypt AES Key with private key
				optiga_status = OPTIGA_LIB_BUSY;
				crypt = optiga_crypt_create(0, optiga_crypt_callback, NULL);
				optiga_status = OPTIGA_LIB_BUSY;
				encryption_scheme = OPTIGA_RSAES_PKCS1_V15;
			

				optiga_status = optiga_crypt_rsa_decrypt_and_export(	crypt,
																															encryption_scheme,																															decrypt,																															sizeof(decrypt),																															NULL,																															0,																															optiga_key_id,																														decrypted,																															&decrypted_length);



![image](https://github.com/Infineon/optiga-trust-m/assets/62071761/81d81289-6055-4968-bce3-e99eeb3a0285)

**Comment by sss551** [comment]
Hi @DIEGGOMEZ ,

Can you please tell which optiga_key_id you are using for decryption ?

Can you please tell if there is a key present in optiga_key_id  OID ?

If there is no key in the OID(slot), please generate RSA key pair using optiga_crypt_rsa_generate_keypair() API ?

Best regards,
Shiva Sai

**Reply by DIEGGOMEZ** [reply]
Hi @sss551 !

Can you please tell which optiga_key_id you are using for decryption ? **E0FD**

Can you please tell if there is a key present in optiga_key_id OID ? Yes., it should be provided, theoretically i cant read it, right?

If there is no key in the OID(slot), please generate RSA key pair using optiga_crypt_rsa_generate_keypair() API ? It should be an optiga key!

Regards!!

**Reply by RaymWong** [reply]
@DIEGGOMEZ 
Noted that **optiga_util_create** must be called before **optiga_util_open_application(util,0)** is called.
see example here:
https://github.com/Infineon/optiga-trust-m/blob/main/examples/optiga/example_optiga_crypt_rsa_decrypt.c

**Reply by DIEGGOMEZ** [reply]
Thank you for the advice, the error persist but appreciate it!

**Reply by koblibri** [reply]
Hi, the "Command out of Sequence" Error should only occur in this case, when there is no RSA Private Key present in the slot. 
By default, RSA Key 0xE0FD is empty. Please check again, if a key is present.

I attached an example that follows the full flow of "open -> generate key -> encrypt data -> decrypt data" which may give you some guidance.
[rsa_test.txt](https://github.com/Infineon/optiga-trust-m/files/15481775/rsa_test.txt)
